File: 성규/11주차/최소직사각형/PGS_Lv1_86491.js
/*
1. 명함에 다 넣기 위해서는 가로와 세로값이 다른 값들을 다 포함할 수 있어야한다. 
2. 가로와 세로는 눕혀서 수납할 수 있다(값변경이 가능하다)
3. 따라서, 가로와 세로에서 큰 값을 한쪽으로 몰아 max값끼리 곱하면 결과가 나온다
*/
function solution(sizes) {
  const weight = [];
  const height = [];

  sizes.map((item) => {
    const [w, h] = item;
    if (w > h) {
      weight.push(w);
      height.push(h);
    } else {
      weight.push(h);
      height.push(w);
    }
  });
  return Math.max.apply(null, weight) * Math.max.apply(null, height);
}

File: 수빈/src/PGS_Lv1_86491.java
/**
 * 프로그래머스 최소직사각형 / Lv1 / 5분
 */

public class PGS_Lv1_86491 {

    public static int solution(int[][] sizes) {
        int wMax = 0;
        int hMax = 0;

        for (int[] size : sizes) {
            int w = Math.max(size[0], size[1]);
            int h = Math.min(size[0], size[1]);

            wMax = Math.max(wMax, w);
            hMax = Math.max(hMax, h);
        }

        return wMax * hMax;
    }
}
